src/crypto/CHIPCryptoPAL.h

```diff
@@ -565,23 +565,28 @@ class P256Keypair : public P256KeypairBase
 using Symmetric128BitsKeyByteArray = uint8_t[CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
 
 /**
- * @brief Platform-specific AES key
+ * @brief Platform-specific Symmetric key handle
  *
- * The class represents AES key used by Matter stack either in the form of raw key material or key
+ * The class represents a key used by the Matter stack either in the form of raw key material or key
  * reference, depending on the platform. To achieve that, it contains an opaque context that can be
  * cast to a concrete representation used by the given platform. Note that currently Matter uses
  * 128-bit symmetric keys only.
+ *
+ * @note Symmetric128BitsKeyHandle is an abstract class to force child classes for each key handle type.
+ *       Symmetric128BitsKeyHandle class implements all the necessary components for handles.
+ *       Child classes only need to implement a constructor, implement a destructor and delete all the copy operators.
  */
-class Aes128KeyHandle
+class Symmetric128BitsKeyHandle
 {
 public:
-    Aes128KeyHandle() = default;
-    ~Aes128KeyHandle() { ClearSecretData(mContext.mOpaque); }
+    Symmetric128BitsKeyHandle() = default;
+    // Destructor is implemented in the .cpp. It is pure virtual only to force the class to be abstract.
+    virtual ~Symmetric128BitsKeyHandle() = 0;
```

Comment:
We don't really need a virtual destructor, do we? I think we should rather make it protected non-virtual to reduce memory footprint.

Reply:
Good catch, updated in #30751